activerecord/lib/active_record/connection_adapters/postgresql_adapter.rb

```diff
@@ -667,6 +682,9 @@ def configure_connection
             execute("SET time zone '#{@local_tz}'", 'SCHEMA')
           end
 
+          # Set interval output format to ISO 8601 for ease of parsing by ActiveSupport::Duration.parse
+          execute('SET intervalstyle = iso_8601', 'SCHEMA')
```

Comment:
It's a shame that the Ruby PG gem (and libpq) don't allow for receiving certain columns in binary, but others as text. `interval` is significantly easier to deal with in the binary form (64 bit signed integer representing number of microseconds, 32 bit signed integer representing number of days, 32 bit signed integer representing number of months).